src/main/java/org/tuiter/controllers/UserController.java

```diff
@@ -83,21 +83,15 @@ public void setUserService(UserServiceImpl userService) {
 			produces = MediaType.APPLICATION_JSON_VALUE,
 			consumes = MediaType.APPLICATION_JSON_VALUE
 			) 
-	public ResponseEntity<HttpStatus> edit(@RequestBody EditUserBean body) {
-		User user = userService.findByUsername(body.getUsername());
+	public ResponseEntity<HttpStatus> edit(@Valid @RequestBody EditUserBean body) {
+		User user = userService.findByUsername(body.getRequester());
 		
 		if (user == null) {
 			throw new TuiterApiException("User not found!", HttpStatus.INTERNAL_SERVER_ERROR, ErrorCode.NOT_FOUND);			
 		}
 		
-		if (!body.getName().isEmpty()) {
-			user.setName(body.getName());
-		}
-		
-		if (!body.getPhotoUrl().isEmpty()) {
-			user.setPhotoUrl(body.getPhotoUrl());
-		}
-		
+		user.setName(body.getName());
```

Comment:
the logic should be made in a service class, controller should be responsible just for routing and delegation.